File: test-fixtures/codebase/Sample.swift
/// Sample Swift module for testing.
///
/// This module demonstrates various Swift features.

import Foundation

/// A greeter struct that holds a name.
/// Used for generating greeting messages.
@available(iOS 13.0, macOS 10.15, *)
public struct Greeter {
    private let name: String

    /// Creates a new Greeter instance.
    public init(name: String) {
        self.name = name
    }

    /// Returns a greeting message.
    public func greet() -> String {
        return "Hello, \(name)!"
    }

    private func privateMethod() -> String {
        return name
    }
}

/// Private helper struct.
struct PrivateHelper {
    let value: Int

    func calculate(x: Int) -> Int {
        return value + x
    }
}

/// Add two numbers together.
/// - Parameters:
///   - a: First number
///   - b: Second number
/// - Returns: The sum of a and b
@inlinable
public func add(_ a: Int, _ b: Int) -> Int {
    return a + b
}

private func privateFunction() -> Int {
    return 42
}

/// Process data with validation.
@discardableResult
public func processData(_ data: String) -> String? {
    guard !data.isEmpty else {
        return nil
    }
    return data.uppercased()
}

/// Public protocol for services.
public protocol Service {
    /// Executes the service.
    func execute()
}

/// Observable data processor class.
@MainActor
public class DataProcessor {
    /// Processes input data.
    @available(*, deprecated, message: "Use processV2 instead")
    public func process(_ data: String) {
        print(data)
    }
}

File: test-fixtures/codebase/sample.dart
/// Sample Dart library for testing.
///
/// This library demonstrates various Dart features.
library sample;

/// A greeter class that holds a name.
/// Used for generating greeting messages.
class Greeter {
  final String _name;

  /// Creates a new Greeter instance.
  Greeter(this._name);

  /// Returns a greeting message.
  String greet() {
    return 'Hello, $_name!';
  }

  void _privateMethod() {
    // Private method
  }
}

/// Private helper class (underscore prefix).
class _PrivateHelper {
  final int value;

  _PrivateHelper(this.value);

  int calculate(int x) {
    return value + x;
  }
}

/// Add two numbers together.
/// Returns the sum of a and b.
@pragma('vm:prefer-inline')
int add(int a, int b) {
  return a + b;
}

int _privateFunction() {
  return 42;
}

/// Process data with validation.
@Deprecated('Use processDataV2 instead')
String? processData(String data) {
  if (data.isEmpty) {
    return null;
  }
  return data.toUpperCase();
}

/// Public interface for services.
abstract class Service {
  /// Executes the service.
  void execute();
}

/// Data processor mixin.
@pragma('vm:entry-point')
mixin DataProcessorMixin {
  /// Processes the given input.
  void process(String input) {
    print(input);
  }
}
